app/scenes/Document/components/DataLoader.tsx

```diff
@@ -73,6 +75,19 @@ function DataLoader({ match, children }: Props) {
     fetchDocument();
   }, [ui, documents, document, shareId, documentSlug]);
 
+  React.useEffect(() => {
+    async function fetchIntegrations() {
+      try {
+        integrations.fetchPage({
```

Comment:
I still think in the long run it might be better to just pull all these in on app load, otherwise there's a more likely race between this request completing and the document content showing.

Reply:
What's the right place? `app/components/Editor.tsx`? Any pointers?

Reply:
I'm not sure without checking but they don't _need_ to be required.

Reply:
Somewhere super early in the app-boot that doesn't exist yet 😄 – here is okay for this PR, but lets make sure it doesn't load integrations every time, it should only load if they haven't been already.